sound/soc/intel/boards/sof_cs42l42.c

```diff
@@ -677,8 +633,6 @@ static int sof_audio_probe(struct platform_device *pdev)
 	if (ret)
 		return ret;
 
-	ctx->common_hdmi_codec_drv = mach->mach_params.common_hdmi_codec_drv;
-
```

Comment:
this patch is a bit hard to review, maybe split it in two with the removal of the hdac_hdmi stuff and then introduction of common structures?